Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/EntityPropertyEditor.cpp

```diff
@@ -797,7 +797,7 @@ namespace AzToolsFramework
     {
         if (IsEntitySelected(entityId))
         {
-            UpdateContents(); // immediately refresh
+            QueuePropertyRefresh();
```

Comment:
I went back in time and found this was added back in 2017 in order to fix an issue where pinned inspectors still showed the entity after a slice was deleted. So I think it is safe to do this. The `QueuePropertyRefresh` method already immediately disables actions/edits in the component editors and then queues the call to `UpdateContents`.

As for an automated test, it might even be better to create a benchmark for this workflow so we can make sure you can undo a lot of these changes and verifying it happens within a certain threshold.

Reply:
Thanks very much for doing the detective work @cgalvan! Definitely be something to keep an eye on and a test+benchmark are worth adding. @chiyenteng could we add a follow-up GHI for this?

Reply:
Oh sorry I see you did this below 😅 Thanks!